### PR TITLE
feat: seperate PR size types: additions, deletions and total

### PR DESCRIPTION
**Pull request by gangachris**
This PR adds the ability to specify PR size types. 
```yml
  - do: size
    lines:
      total:
        count: 500
        message: Change is very large. Should be under 500 lines of additions and deletions.
      additions:
        count: 250
        message: Change is very large. Should be under 250 lines of additions 
      deletions:
        count: 500
        message: Change is very large. Should be under 250 lines of deletions.
```

### CODE REVIEW

**Comment by sergiughf** [comment]
This is awesome! Can we please get it reviewed and incorporated?:) :+1:

File: docs/README.md [hunk 1]
```diff
@@ -163,15 +163,20 @@ The above will validate that both the files `package-lock.json` and `yarn.lock`
 ### Size
 
 `size` validates that the size of changes in the pull request conform to a
-specified limit. Currently this only lets you validate that the total number of
-changed lines is below a certain amount using the `max` option:
+specified limit. We can pass in three options: `total`, `additions` or `deletions`. Each of this take in a `count` and `message`.
```
[the remark this answers is not in the document]
Comment (gangachris) [reply docs/README.md]: Yeah, I think having it as an alias would be nice. Thanks.

File: lib/validators/size.js [hunk 2]
```diff
@@ -75,22 +75,69 @@ class Size extends Validator {
...
+
+      output.push(constructOutput(
+          validatorContext,
+          inputMessage,
+          validationSettings,
+          result
+        ))
+    }
+
+    if (!!validationSettings.lines.total && !!validationSettings.lines.total.count) {
```
[the remark this answers is not in the document]
Comment (gangachris) [reply lib/validators/size.js]: https://github.com/mergeability/mergeable/pull/229/commits/d9b66497f3418087e0cb6498716d01bf17428d19

File: lib/validators/size.js [hunk 3]
```diff
@@ -75,22 +75,69 @@ class Size extends Validator {
...
     }
-    const output = constructOutput(
-      validatorContext,
-      inputMessage,
-      validationSettings,
-      result
-    )
 
-    return consolidateResult([output], validatorContext)
+    return consolidateResult(output, validatorContext)
```
[the remark this answers is not in the document]
Comment (gangachris) [reply lib/validators/size.js]: https://github.com/mergeability/mergeable/pull/229/commits/d9b66497f3418087e0cb6498716d01bf17428d19

File: docs/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gangachris) [reply docs/README.md]: https://github.com/mergeability/mergeable/pull/229/commits/ac639170ea6a1cd2daf57e5f7795407f2237123d

**Comment by weyert** [comment]
Looking forward to this PR being merged :D

**Comment by weyert** [comment]
I  haven't tested it but I can test it tomorrow?

**Comment by weyert** [comment]
I am not sure if this working. I have applied to my own forked bot, which is just this repo with this and another PR merged plus a `Dockerfile`. While I have a bunch of validations defined in the yml it keeps saying only one validation had to be run. I am not sure if this is because of the script error I am experiencing when using #170. I will probably revert that PR and try it without.

**Comment by shine2lay** [comment]
@weyert I just tested it, and it seems to be working great! you can see it in my testRepo here : https://github.com/shine2lay/MergeableTest/pull/49/checks?check_run_id=211125791

**Comment by weyert** [comment]
I can confirm it's working :)

**Comment by shine2lay** [comment]
@weyert thanks, i'll merge it in now

**Comment by shine2lay** [comment]
@sergiughf this PR has been merged! :tada:

**Pull request merged**